ldb/block.go

```diff
@@ -290,3 +290,41 @@ func (db *LevelDb) NewestSha() (rsha *btcwire.ShaHash, rblkid int64, err error)
 
 	return &sha, db.lastBlkIdx, nil
 }
+
+// fetchAddrIndexTip returns the last block height and block sha to be indexed.
+// Meta-data about the address tip is currently cached in memory, and will be
+// updated accordingly by functions that modify the state. This function is
+// used on start up to load the info into memory. Callers will use the public
+// version of this function below, which returns our cached copy.
+func (db *LevelDb) fetchAddrIndexTip() (*btcwire.ShaHash, int64, error) {
+	db.dbLock.Lock()
+	defer db.dbLock.Unlock()
+
+	data, err := db.lDb.Get(addrIndexMetaDataKey, db.ro)
+	if err != nil {
+		return &btcwire.ShaHash{}, -1, btcdb.ErrAddrIndexDoesNotExist
```

Comment:
Very good point, returning nil instead does seem to be better behavior. It currently returns a pointer to the zero hash simply to emulate the behavior of `NewestSha`, since they are rather similar. 

If changed to return nil, then following the same logic, should we also change `NewestSha` and update all callers the react to the new behavior? I think that'd require changes in both `btcd` and `btcchain`, and maybe some other packages I can't name off-head.